tools/analyzer_plugin/lib/src/diagnostic/arrow_function_prop.dart

```diff
@@ -21,6 +23,10 @@ class ArrowFunctionPropCascadeDiagnostic extends ComponentUsageDiagnosticContrib
       if (rhs is FunctionExpression && rhs.body is ExpressionFunctionBody) {
         final body = rhs.body as ExpressionFunctionBody;
 
+        // If a cascade expression is not found in the body, it is not an un-parenthesized
+        // function expression in the middle of another cascade... do not lint.
+        if (allDescendantsOfType<CascadeExpression>(body).isEmpty) continue;
```

Comment:
Could this incorrectly avoid adding this lint to something like this snippet below? There's probably not a straightforward way to differentiate this case from cases where we want to lint, though.
```
        ..foo = () => FooObject()..someMethod()
```

Reply:
@corwinsheahan-wf in offline discussions - we decided that linting this is a good idea anyway - because one formatted - cascades like that without parentheses wrapping around the entire expression... makes stuff extremely difficult to read.

For example... I've seen quite a bit of stuff like this - which makes me want to 🤮 

```dart
@override
render() {
  return (Foo()
    ..id = 'foo'
    ..className = 'dont-mess-with-foo'
    ..barRenderer = () => (Bar()
      ..id = 'bar'
      ..aria.label = 'Yeah I put my feet on Foos couch!'
    )(
      'I am a bar',
    )
  )(); 
}
```

Once wrapped - the nested function is much easier to reason about, IMO:

```dart
@override
render() {
  return (Foo()
    ..id = 'foo'
    ..className = 'dont-mess-with-foo'
    ..barRenderer = (() =>
        (Bar()
          ..id = 'bar'
          ..aria.label = 'Yeah I put my feet on Foos couch!'
        )(
          'I am a bar',
        )
    )
  )();
}
```

Reply:
I like it